### PR TITLE
updated env template and removed GRAPH_DB_TYPE

### PR DESCRIPTION
**Pull request by tushar1245**
## Description

updated the env template. It had older default values and also removed the use of redundant key from dockerfiles (GRAPH_DB_TYPE).
message broker defaulted to redis in helm chart and made kafka and zookeeper pods to deploy conditionally

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

## Summary by CodeRabbit

* **Configuration**
  * Simplified graph database selection to use a single setting.
  * Updated deployment templates and installation setup to remove redundant graph database options.
  * Redis is now the default key-value store and message broker.
  * Added explicit Neo4j connection settings and refreshed deployment configuration.
  * Kafka and ZooKeeper resources are now deployed only when Kafka is selected.

### CODE REVIEW

File: backend/env.template [hunk 1]
```diff
@@ -20,20 +20,37 @@ EMBEDDING_SERVER_ALLOW_REMOTE_CODE=false
...
 
-KAFKA_BROKERS=localhost:9092
+NEO4J_URI=bolt://localhost:7687
+NEO4J_USERNAME=neo4j
+NEO4J_PASSWORD=your_neo4j_password
+NEO4J_DATABASE=neo4j
+
+
+# Message broker: "kafka" or "redis" (uses Redis Streams)
+MESSAGE_BROKER=redis
```
[the remark this answers is not in the document]
Comment (tushar1245) [reply backend/env.template:51]: updated the default to redis in helm chart

File: deployment/helm/pipeshub-ai/templates/deployment.yaml [hunk 2]
```diff
@@ -169,13 +167,15 @@ spec:
...
-              value: {{ .Values.messageBroker.type | default "kafka" | quote }}
+              value: {{ .Values.messageBroker.type | default "redis" | quote }}
             - name: REDIS_STREAMS_MAXLEN
               value: {{ .Values.messageBroker.redisStreamsMaxLen | default "500000" | quote }}
 
+            {{- if eq .Values.messageBroker.type "kafka" }}
             # Kafka config
             - name: KAFKA_BROKERS
               value: "{{ include "pipeshub-ai.fullname" . }}-kafka:9092"
+            {{- end }}
```
[the remark this answers is not in the document]
Comment (tushar1245) [reply deployment/helm/pipeshub-ai/templates/deployment.yaml:178]: fixed, made redis as default mesage broker

File: backend/nodejs/apps/src/libs/constants/messaging.constants.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ export const ENV_MESSAGE_BROKER = 'MESSAGE_BROKER';
 export const ENV_REDIS_STREAMS_MAXLEN = 'REDIS_STREAMS_MAXLEN';
 export const ENV_REDIS_STREAMS_PREFIX = 'REDIS_STREAMS_PREFIX';
 
-export const DEFAULT_MESSAGE_BROKER = MessageBrokerType.KAFKA;
+export const DEFAULT_MESSAGE_BROKER = MessageBrokerType.REDIS;
```
[the remark this answers is not in the document]
Comment (tushar1245) [reply backend/nodejs/apps/src/libs/constants/messaging.constants.ts:8]: it is fine

File: backend/nodejs/apps/src/modules/configuration_manager/config/config.ts [hunk 4]
```diff
@@ -31,8 +31,8 @@ export const getHashedSecretKey = (): string => {
 
 export const loadConfigurationManagerConfig =
   (): ConfigurationManagerConfig => {
-    // Determine store type from KV_STORE_TYPE env variable (defaults to etcd)
-    const kvStoreType = process.env.KV_STORE_TYPE?.toLowerCase() || 'etcd';
+    // Determine store type from KV_STORE_TYPE env variable (defaults to redis)
+    const kvStoreType = process.env.KV_STORE_TYPE?.toLowerCase() || 'redis';
```
[the remark this answers is not in the document]
Comment (tushar1245) [reply backend/nodejs/apps/src/modules/configuration_manager/config/config.ts:35]: can be ignored

**Review by abhishek-pipeshub — APPROVED** [review]
